gcc/testsuite/rust/compile/issue-2650-2.rs

```diff
@@ -0,0 +1,6 @@
+// { dg-additional-options "-frust-edition=2015" }
+
+// { dg-excess-errors "'async fn' is not permitted in Rust 2015" }
```

Comment:
You won't be able to match the backtick token using dejaGNU because AFAIK it is based on the locale. User a dot `.` to match any character instead.